[WordPress → Jetpack app branding] Update app promo in Reader

SiobhyB opened the pull request:
_Context:_ Following the plans to [refocus the WordPress apps on core features](https://make.wordpress.org/mobile/2022/07/27/refocusing-the-wordpress-app-on-core-features/), we plan to move features that are specific to WordPress.com to the Jetpack app. As such, we will eventually replace all mentions of the WordPress app in Calypso with Jetpack.

#### Proposed Changes

The main focus of this PR involves updating the mobile app promo in the Reader's sidebar to mention the Jetpack, rather than the WordPress, mobile app. This also has the side effect of slightly changing the desktop app promo's appearance, as a shared component is used to promote both the desktop and mobile.   

<details>
 <summary>Code notes ⤵️</summary>

* The `jetpack/app-branding` feature flag (added in https://github.com/Automattic/wp-calypso/pull/66814) is used to only display the new Jetpack app mentions when the flag is set to true (currently only for wpcalypso and development builds).
* Note, the feature flag, along with any code related to the WordPress app, will be removed when we're ready to deploy all web-related changes as part of the Jetpack app migration.
* This PR also includes some light refactoring of the HTML and CSS for the `app-promo` component, to make sure of `display: flex` for positioning elements, rather than `position: absolute`. This was done to make it easier to match the designs for promoting the mobile app.
* As the `app-promo` component is used to promote both the mobile and the desktop app, these changes modifies the design for both promotions. The screenshots below show all before/after states.
</details>

#### Screenshots

<details>
 <summary>Design comparison ⤵️</summary>

| Design  | Actual |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192336752-f8504e0e-41ef-450b-93e4-4773af5dabf9.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194160684-05a620f0-2abb-4825-94e3-5f8cf0b8020b.png" width="100%"> |
</details>

<details>
 <summary>Mobile app promo before/after ⤵️</summary>

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192349308-aa14a9ab-1026-4df4-82c7-d8d67325b97d.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194160684-05a620f0-2abb-4825-94e3-5f8cf0b8020b.png" width="100%"> |
</details>

<details>
 <summary>Desktop app promos before/after ⤵️</summary>

As this change involved some changes to the promo, the design for the desktop app promo has been tweaked slightly:

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192349153-58675fd1-c268-466a-8755-059f6f76b769.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194160009-822bf9e6-0f57-42b7-ae08-8ad5ed13157d.png" width="100%"> |

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192348982-c2095b68-0602-4222-b465-a52ce6975c42.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194160153-86ec3754-e9e6-4e10-91f8-b834349bc855.png" width="100%"> |

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192349896-39621899-4af9-4d41-a7e3-fee5acd0ea98.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194160494-d6eb96ad-afd5-48a1-b97e-f55329a7c7b9.png" width="100%"> |

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192349592-7b4b0170-d854-413b-a51f-01461b635c37.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194160962-058f4897-8e6a-4209-8daa-62ec9080543f.png" width="100%"> |

</details>

<details>
 <summary>Different colour schemes ⤵️</summary>

**Ectoplasm**

| Desktop  | Mobile |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/194352818-b4390a15-4bb4-48e9-8f2b-30a3cb70de86.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194352676-c261cdfa-7513-4cc5-b559-669a2aa3c0df.png" width="100%"> |

**Nightfall**

| Desktop  | Mobile |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/194352461-48a429cc-811d-415e-94de-566743483ef5.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194352280-4935e4fc-9a25-451c-835a-c04d64dddc13.png" width="100%"> |

**Ocean**

| Desktop  | Mobile |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/194353171-f282eee6-baaf-464d-bbd7-93ebddb99d30.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194353356-663c1c56-bbed-442f-859b-10c897d1e92c.png" width="100%"> |

**Sunrise**

| Desktop  | Mobile |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/194353779-2ede12d8-0946-44d2-8fc3-8ab38d4e47fb.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194354300-6f8d2d09-e576-42f3-9133-75831cf3bbe1.png" width="100%"> |

**Sunset**

| Desktop  | Mobile |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/194352077-134f67af-fd4a-4bca-a91c-b40ded91527c.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194354443-97b4fec0-e538-409b-b045-d55a3153c98e.png" width="100%"> |

</details>


#### Testing Instructions

* Navigate to the Reader in an account where you haven't dismissed the sidebar app promo. (You can manually change `shouldRenderAppPromo` to `true` [here](https://github.com/Automattic/wp-calypso/blob/dded2cf2a4d3d3770818998875bd4c3616bd93f7/client/reader/sidebar/promo.jsx#L17) if testing locally.)
* Reload the page to cycle through the various app promos to confirm their appearance matches the above screenshots.

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

SiobhyB committed:
Set up 'displayJetpackAppBranding' conditional

This conditional will be used to set some changes behind a feature flag, before the Jetpack app is advertised for all users.

SiobhyB committed:
Display Jetpack logo when advertising Jetpack app

With this commit, the 'displayJetpackAppBranding' conditional is used to display a different logo in scenarios where we're advertising the Jetpack app.

For this purpose, the new Jetpack logo that we're using has been added to the 'wp-to-jp-vertical.svg' file.

SiobhyB committed:
Display and style message promoting Jetpack app

SiobhyB committed:
Apply custom style to Jetpack app promo button

SiobhyB committed:
Refactor app promo's HTML/CSS for easier centring

With this commit, the HTML/CSS for the app promo is refactored to make use of flex box, making it easier to centre elements within the promo.

tyxla commented on client/blocks/app-promo/index.jsx:
```diff
@@ -148,29 +153,36 @@ export class AppPromo extends Component {
+		const message = displayJetpackAppBranding ? (
+			<p className="app-promo__paragraph">
+				<a href="https://wp.com/app" className="app-promo__jetpack-app-link">
+					Get the Jetpack app
```
Should we localize this string?

SiobhyB replied:
The changes in https://github.com/Automattic/wp-calypso/pull/68293#discussion_r981374638, following [this other comment](https://github.com/Automattic/wp-calypso/pull/68293#discussion_r981251709) should address this.

tyxla commented on client/blocks/app-promo/index.jsx:
```diff
@@ -148,29 +153,36 @@ export class AppPromo extends Component {
+			<p className="app-promo__paragraph">
+				<a href="https://wp.com/app" className="app-promo__jetpack-app-link">
+					Get the Jetpack app
+				</a>{ ' ' }
```
Is there any chance to style this without the need to add an extra inline space?

SiobhyB replied:
The changes in https://github.com/Automattic/wp-calypso/pull/68293#discussion_r981374638, following [this other comment](https://github.com/Automattic/wp-calypso/pull/68293#discussion_r981251709) should address this.

tyxla commented on client/blocks/app-promo/index.jsx:
```diff
@@ -148,29 +153,36 @@ export class AppPromo extends Component {
+				<a href="https://wp.com/app" className="app-promo__jetpack-app-link">
+					Get the Jetpack app
+				</a>{ ' ' }
+				{ translate( 'to use Reader anywhere, any time.' ) }
```
Splitting strings like this can be problematic in some languages. "Get the Jetpack app to use Reader ..." should be a single string, and we can use the `translate()` componentization features to achieve that - see this for example:

https://github.com/Automattic/wp-calypso/blob/f2e6f88ae65c8571bbac21b284c29b48ec987598/apps/notifications/src/panel/templates/body.jsx#L134

SiobhyB replied:
I appreciate you making me aware of this. 🙇‍♀️ I've gone ahead to implement this in https://github.com/Automattic/wp-calypso/pull/68293/commits/df7df25dfaa33f8dc37230eb1a331b1b266e166b.

tyxla commented on client/blocks/app-promo/index.jsx:
```diff
@@ -148,29 +153,36 @@ export class AppPromo extends Component {
+				{ translate( 'to use Reader anywhere, any time.' ) }
+			</p>
+		) : (
+			<p className="app-promo__paragraph">
```
Seems like in both cases we render a `<p className="app-promo__paragraph" />`. Should the conditional logic be inside the `<p>` then?

SiobhyB replied:
Ah, good point! I've made that change in https://github.com/Automattic/wp-calypso/pull/68293/commits/edf541644a1c27bf97da13f594cccf52a40fda77.

tyxla commented on client/blocks/app-promo/index.jsx:
```diff
@@ -148,29 +153,36 @@ export class AppPromo extends Component {
-						width="32"
-						height="32"
+						src={ displayJetpackAppBranding ? jetpackLogoImage : wordpressLogoImage }
+						width={ displayJetpackAppBranding ? '25' : '32' }
```
Should the dimensions be numbers and not strings?

SiobhyB replied:
It looks like either can be used, I've gone ahead to update to numbers in https://github.com/Automattic/wp-calypso/pull/68293/commits/6a8513039095417c302c6f4a901325cb7b9fa412 as that seems more 'proper'. :)

tyxla commented on client/blocks/app-promo/index.jsx:
```diff
@@ -185,6 +197,14 @@ export class AppPromo extends Component {
 						) }
 					</p>
 				</Dialog>
+				<Button
+					tabIndex="0"
+					className="app-promo__dismiss"
+					onClick={ this.dismiss }
+					aria-label={ translate( 'Dismiss' ) }
+				>
+					<Gridicon icon="cross" size={ 16 } />
+				</Button>
```
This looks repeated from above. Should we abstract it to a function or constant so we don't repeat it twice?

SiobhyB replied:
Thanks for the suggestion, I agree that's a good idea! I've gone ahead to implement that in https://github.com/Automattic/wp-calypso/pull/68293/commits/a9db374d168efbd01b9f9bf2ce087c86910da5b1.

tyxla commented on client/blocks/app-promo/style.scss:
```diff
@@ -1,46 +1,35 @@
 
 	svg {
-		fill: var(--color-neutral-10);
+		fill: #616161;
```
Any chance to use a color variable here?

SiobhyB replied:
`#616161` and `#1e1e1e` don't have existing variables associated with them in the Calypso codebase and aren't part of [the Color Studio](https://color-studio.blog/) listing. Would you perhaps suggest still creating variables at the top of this individual stylesheet, or do you think this is fine to overlook in this instance?

tyxla replied:
We can always hardcode a color, but that usually creates problems when the user (or the app) uses a different color scheme that overrides the default colors. Then the hardcoded color might not quite fit into the color scheme and might look odd to the user 😉 Is there a way we can instead pick a variable that is close enough that will do the job?

jsnajdr replied:
Other buttons with a "cross" icon use `--color-text-subtle`, inherited from `.button.is-borderless`. If there is no strong reason for the App Promo UI to have a truly unique look, it should reuse colors from the surrounding Calypso and the current color theme.

SiobhyB replied:
Thanks for elaborating further, both. I've gone ahead to replace `#616161` with `--color-text-subtle` (`#50575e` by default).

shaunandrews replied:
For future reference, if there are semantic (and similar) colors that exist currently, we should likely always use those colors — often the design doesn't take those into consideration or mistakenly uses hardcoded values.

tyxla commented on client/blocks/app-promo/style.scss:
```diff
@@ -50,6 +39,32 @@
+	color: #1e1e1e;
+
+	&:hover {
+		color: #1e1e1e;
```
Any chance to use a color variable?

SiobhyB replied:
Replied [here](https://github.com/Automattic/wp-calypso/pull/68293#discussion_r981396905).

SiobhyB replied:
Following the [discussion here](https://github.com/Automattic/wp-calypso/pull/68293#discussion_r981396905), I've gone ahead to replace `#1e1e1e` with `--color-text` (`#2c3338` by default).

tyxla reviewed (commented):
Nice work here 👍 I've left some suggestions, let me know what you think.

SiobhyB committed:
Make 'dismissButton' function to avoid duplication

SiobhyB committed:
Use numbers (not strings) to define dimensions

SiobhyB committed:
Ensure link text is correctly translated

It was mentioned here that splitting strings can be problematic: https://github.com/Automattic/wp-calypso/pull/68293#discussion_r981251709

As such, this commit uses the "translate" function's componentization feature to join the link in this paragraph up with the main text. It'll now be rendered as a full string, which will be easier for translators to handle.

SiobhyB committed:
Refactor conditional markup to avoid duplication

As '<p className="app-promo__paragraph">' is used regardless of whether 'displayJetpackAppBranding' is true or false, the conditional code related to 'displayJetpackAppBranding' can be rendered inside that paragraph markup. This helps to avoid some duplication.

SiobhyB committed:
Avoid console error as a result of empty function

tyxla reviewed (commented):
Nothing else to add here on the code side, besides the color variables we already talked about 👍 Looking good!

Since I'm not involved with the project I'll leave the final review and approval to your team.

SiobhyB committed:
Replace hex codes with color variables

SiobhyB committed:
Optimize SVG for performance

With this commit, I used an online tool to reduce the SVG, for better performance.

shaunandrews commented:
This looks nice.

I wasn't aware of the callouts for the desktop app. Can we update them to use the same visual language?

<img width="368" alt="CleanShot 2022-09-30 at 14 41 42@2x" src="https://user-images.githubusercontent.com/191598/193335913-a56f5475-b383-48d7-87d4-158604af5b30.png">

SiobhyB commented:
> I wasn't aware of the callouts for the desktop app. Can we update them to use the same visual language?

@shaunandrews, yes, absolutely. I wanted to check in about the blue colour. There isn't a matching variable for `#3F5BE0`, so I've gone with the `--color-link` variable, which uses `#0675c4`. This clashes a little with the icon's colour:

<img width="270" alt="Screenshot 2022-10-04 at 14 04 29" src="https://user-images.githubusercontent.com/2998162/193826552-eabef297-eb06-4b0c-8121-25ef3bb4b1a1.png">

Does the above screenshot seem okay to you or would you like to change the colours to match more closely? If you'd like to change the colours, is there another colour from the [list of variables here](https://github.com/Automattic/wp-calypso/blob/73abf41a2325ae3a1947f916648faa6e1e6fe967/packages/calypso-color-schemes/src/shared/color-schemes/_default.scss#L210-L239) you'd prefer to see for the blue or should I update the svg to match the link as follows?

<img width="277" alt="Screenshot 2022-10-04 at 14 07 39" src="https://user-images.githubusercontent.com/2998162/193827188-d0579239-0d9e-472a-96d8-1cde68782bca.png">

dcalhoun commented on client/blocks/app-promo/index.jsx:
```diff
@@ -138,8 +146,9 @@ export class AppPromo extends Component {
 						height="32"
 						alt="WordPress Desktop Icon"
```
I wonder if this `alt` is valuable. IMO it is mostly noise when navigating this button group with a screen reader. WDYT about using an empty `alt` attribute?

SiobhyB replied:
I appreciate you keeping accessibility in mind, thank you! I've gone ahead to replace this with an empty alt tag in https://github.com/Automattic/wp-calypso/pull/68293/commits/4a9fa624285f54c193fcd488dd859fda5e026efa.

dcalhoun commented on client/blocks/app-promo/index.jsx:
```diff
@@ -148,29 +157,31 @@ export class AppPromo extends Component {
+						src={ displayJetpackAppBranding ? jetpackLogoImage : wordpressLogoImage }
+						width={ displayJetpackAppBranding ? 25 : 32 }
+						height={ displayJetpackAppBranding ? 43 : 32 }
 						alt="WordPress App Icon"
```
Similar to a comment on the desktop app `alt` text, I wonder how beneficial this `alt` value is. WDYT about using an empty string instead? 

If we do keep `alt` text here, we need to dynamically mention "Jetpack" instead of "WordPress".

SiobhyB replied:
Addressed as part of [this comment](https://github.com/Automattic/wp-calypso/pull/68293#discussion_r988345981). 🙇‍♀️

dcalhoun commented on client/blocks/app-promo/index.jsx:
```diff
@@ -148,29 +157,31 @@ export class AppPromo extends Component {
-				<Button
-					onClick={ this.sendMagicLink }
-					className="app-promo__link"
+					onClick={ ! displayJetpackAppBranding ? this.sendMagicLink : undefined }
```
I believe we need to add an `href` here for the button group. Without the `href`, this element is clickable, but does not result in any action occurring. It is particularly a big issue when navigating via a screen reader, where it is not possible to click the nested anchor element, so this element is inoperable. 

The following is merely for demonstrative purposes, it may make sense to assign `"https://wp.com/app"` to a constant. 

```suggestion
					onClick={ ! displayJetpackAppBranding ? this.sendMagicLink : undefined }
					href={ ! displayJetpackAppBranding ? "https://wp.com/app" : undefined }
```

SiobhyB replied:
@dcalhoun, thank you for pointing this out! Following your feedback, along with the [design requests from Shaun](https://github.com/Automattic/wp-calypso/pull/68293#issuecomment-1263900484), I've made a few changes to the way the mobile app promo works:

* I've removed the magic link code (in https://github.com/Automattic/wp-calypso/pull/68293/commits/59a96728459d970de5e20a50743867e1405922cd) and replaced this `onClick` function with `this.recordClickEvent`, to match the desktop app promo. 
* I've then made use of the component's `getPromoLink` function for the `href`.

Here's what the before/after looks like:

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/194167675-dfa6eedd-eb9b-47b0-86d9-f37024ef02c8.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/194167862-0a819ebf-9249-461f-8b23-477788b3789f.png" width="100%"> |

The desktop and mobile promos are really close code-wise now, so we could probably merge them into one component when it comes time to remove the Jetpack app's feature flag. I didn't do this now as I thought all the conditionals may make things overly confusing to parse. Let me know if you have other thoughts on this approach, though!

dcalhoun reviewed (changes requested):
This looks good so far. I left a few comments. Some are optional, but some we likely need to address before merging to make sure we appropriately reference Jetpack and support screen readers.

SiobhyB committed:
Remove unnecessary alt tags to improve icon a11y

As the icons being used as purely decorative, it's not necessary to include alt text for screen readers. As such, they're removed with this commit and empty alt tags are used instead.

SiobhyB committed:
Add and style CTA in desktop app button

This commit is the first step towards acting on the design feedback provided here: https://github.com/Automattic/wp-calypso/pull/68293#issuecomment-1263900484

More specifically, the CTA within the desktop app button's copy is added within span tags, and also styled.

SiobhyB committed:
Remove "send email login" code

This code is no longer going to be used in future iterations of the promo.

SiobhyB committed:
Refactor mobile and desktop promos to match design

As the design for the mobile and desktop promos more closely match than before, this commit takes the opportunity to refactor the markup surrounding the promos and removes some now unnecessary code.

SiobhyB committed:
Update icon used in desktop promo

SiobhyB committed:
Fix: Add missing parameter to function

SiobhyB committed:
Fix: Prevent tests from failing

The 'getPromoLink' test was previously failing due to the way that function was being called in the main component's file. This commit addresses that.

SiobhyB commented:
I've gone ahead to update the code following the recent design feedback here, and have also updated the screenshots in the PR's description for reference. For now, I've gone with #0675c4 as the colour for the desktop app promo, but this can be easily changed.

dcalhoun commented on client/blocks/app-promo/index.jsx:
```diff
@@ -94,98 +128,71 @@ export class AppPromo extends Component {
-	};
+	dismissButton = () => (
+		<Button
+			tabIndex="0"
```
Nitpick: I am not sure why this has or needs `tabIndex="0"`. From my testing, it has no impact on the ability to navigate this component via keyboard as the element itself is a `button`, which is focusable by default. There may be something I am missing or context of which I am unaware.

SiobhyB replied:
Thanks for spotting that! I didn't see a use for it in my testing either. I wonder if it was necessary prior to some of the changes that have been made to the button's ordering and positioning (it was previously positioned absolutely and above the other elements in the code). Either way, I've gone ahead to remove this in https://github.com/Automattic/wp-calypso/pull/68293/commits/4d3c4d8f94d240bc89573f2de3064160efb7003f. :)

dcalhoun commented on client/blocks/app-promo/index.jsx:
```diff
@@ -94,98 +128,71 @@ export class AppPromo extends Component {
+					'WordPress.com in the palm of your hands — {{span}}download the mobile app{{/span}}.',
+					{
+						components: {
+							span: <span className="app-promo__desktop-app-link" />,
```
Nitpick: given this selector is used in the context of the mobile promo, a more appropriate name might be `app-promo__wp-app-link`.

SiobhyB replied:
Done in https://github.com/Automattic/wp-calypso/pull/68293/commits/e510ddbde8cdc7611ed555160d20ffe26de16aab ✅

dcalhoun commented on client/blocks/app-promo/style.scss:
```diff
@@ -1,59 +1,73 @@
 }
 
-.app-promo .app-promo__icon {
-	position: absolute;
```
I wanted to draw attention to some nuanced interactivity differences with the refactor away from absolute positioning. They are not necessarily incorrect, but there may be opportunity to improve the accessibility of the interactive elements for all users — cursors, keyboards, and screen readers. 

<details><summary>Interactive Element Size Changes</summary>

https://user-images.githubusercontent.com/438664/194336355-cc498d7c-7369-473e-a750-30e35563592e.mov

</details>

| Production | This PR |
| - | - |
| <img width="273" alt="inspection-before" src="https://user-images.githubusercontent.com/438664/194336504-4e825171-3681-42c1-b3fe-21ebbb7dabf5.png"> | <img width="264" alt="inspection-after" src="https://user-images.githubusercontent.com/438664/194336557-89e096f3-9e66-4789-a725-f7ac615c7df9.png"> |

As the above inspection images show, the sizes of the elements changed in the refactor of positioning. One possibility is that we could tweak the flex layout and padding so that the interactive elements more closely resemble previous sizes for cursor users and the focus state is more noticeable for keyboard users<sup>1</sup>. 

<details><summary>Interactive Element Size Proposal</summary>

https://user-images.githubusercontent.com/438664/194336419-2870810a-7068-424d-9379-e27d70817916.mov

</details>

<details><summary>Propose Code Changes Diff</summary>

```diff
diff --git a/client/blocks/app-promo/style.scss b/client/blocks/app-promo/style.scss
index ad90ec6ee8..d356581d77 100644
--- a/client/blocks/app-promo/style.scss
+++ b/client/blocks/app-promo/style.scss
@@ -1,8 +1,9 @@
 .app-promo {
+	align-items: flex-start;
 	display: flex;
 	max-width: 300px;
 	margin: 10px;
-	padding: 10px;
+	padding: 2px;
 	overflow: hidden;
 	background-color: var(--color-surface);
 	box-shadow: 0 0 0 1px rgba(var(--color-neutral-10-rgb), 0.5), 0 1px 2px var(--color-neutral-0);
@@ -15,8 +16,8 @@
 
 .app-promo .app-promo__dismiss {
 	cursor: pointer;
-	padding: 4px;
-	align-items: flex-start;
+	height: auto;
+	padding: 8px;
 
 	svg {
 		fill: var(--color-text-subtle);
@@ -28,6 +29,7 @@
 	height: auto;
 	border-radius: 0;
 	line-height: 1.5;
+	padding: 8px 0px 8px 8px;
 	color: var(--color-text);
 	font-size: $font-body-extra-small;
 	font-weight: 400;
```

</details>

---

1: The current production implementation does not have a noticeable focus state for the primary banner action and has the dismiss action as the first focused element. IMO improving the focus state is an undeniable improvement; the ordering of the primary and dismiss actions is debatable.

SiobhyB replied:
Thank you for this David, I always appreciate the care you put into reviews and your focus on accessibility. I tested out your changes and agree that it feels like a good improvements. I've gone ahead to implement it in https://github.com/Automattic/wp-calypso/pull/68293/commits/679dfd0346009c47b928c3af94777979808caae4. 🙇‍♀️

dcalhoun reviewed (approved):
The latest revision looks good to me. I left a few nitpicks and a proposal regarding interactive element sizing, but neither should be considered blocking.

SiobhyB committed:
Remove unnecessary tabIndex

This tabIndex appears to be no longer needed (possibly following some refactoring of the code). As such, it's removed with this commit.

See here for further details: https://github.com/Automattic/wp-calypso/pull/68293#discussion_r989058941

SiobhyB committed:
Rename class for clarity

The 'app-promo__desktop-app-link' class currently styles both links for the desktop app and the WordPress mobile app. As such, a more appropriate name for now is 'app-promo__wp-app-link'.

See here for more details: https://github.com/Automattic/wp-calypso/pull/68293#discussion_r989061712

SiobhyB committed:
Increase size of 'clickable' area

The feedback here provided suggestions for increasing the amount of 'clickable' space in the promo's button, while also maintaining a visible focus area: https://github.com/Automattic/wp-calypso/pull/68293#discussion_r989108837

SiobhyB commented:
> The --color-link variable seems like the right one. Can you use that for the SVG's fill value, too? You might have to do it with CSS.

Sure thing, I've gone ahead to update the SVG's fill value. 

> It would be good to test various color schemes, as well. Hopefully using the variable means that everything "just works." 🤞

I tested and added a few screenshots under a new `Different colour schemes ⤵️` section in the PR's description. Sunset was the only scheme I spotted that changes the link's colour, let me know if you think the look of this needs to be tweaked in any way.

shaunandrews commented:
Nice work, @SiobhyB! 🚢

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7573133

Thank you @SiobhyB for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7573133).
